andela/mervels-bn-backend
# #168131419 Profile Page Settings

#### What does this PR do?
add user profile settings
#### Description of Task to be completed?
- add user profile get's created when the user is created
- add update user profile by user Id
- add retrieve user profile using user Id
#### How should this be manually tested?
- clone the repo and checkout to this branch `ft-user-profile-168131419`
- Run seeds to add more data into the database `sequelize db:seed: all`
- Run `npm run start:dev`
- head over to postman and use the route:`localhost:3000/api/v1/auth/signup`
- Create a user using the register user endpoint.
- Verify the user then login with the verified user
- get the created user profile using the route:GET: `localhost:3000/api/v1/profile/myprofile`. You need to pass the generated token received in when signing In the user.
- Update the user profile using the route: PATCH:  `localhost:3000/api/v1/profile` by passing any of the fields 
```
{
     "firstName": "CoolKid",
    "lastName": "Kabiswa",
    "birthDate": "1996-12-09",
    "department": "Operations",
    "phoneNumber": "0732063920",
    "language": "Turkish",
    "currency": "Francs",
    "gender": "MALE",
    "location": "Kigali"
}
```
- To run the test `run npm test`
#### Any background context you want to provide?
- You can use an already created and verified user.
#### What are the relevant pivotal tracker stories?
[168131419](https://www.pivotaltracker.com/story/show/168131419)
#### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/18899285/65171724-d3a94900-da4b-11e9-88f8-2e520ed60910.png)
